### PR TITLE
Gofer HTTP integration

### PR DESCRIPTION
**Pull request by konstantinzolotarev**

### CODE REVIEW

**Comment by alextes** [comment]
Oh hey, wow! Amazing! 🙏 

Looks great at a glance, I'll take a closer look soon.

File: docker-compose.yml
Comment (ckoopmann) [inline docker-compose.yml:4]: Just to be sure, if I wanted to run this service directly instead of with docker I would run `gofer agent` right ?

File: src/main.rs
Comment (ckoopmann) [inline src/main.rs]: Just for my understanding, what does this assert / check exactly ?

File: src/main.rs
Comment (ckoopmann) [inline src/main.rs]: Very nice, definetely cleaner than my `cat testdata` "mock". 
Feel free to delete the `test_data/input.json` since it does not seem to be used anywhere else I think.

File: src/price_provider/gofer/mod.rs
Comment (ckoopmann) [inline src/price_provider/gofer/mod.rs]: Not sure if we really need to make the pair configurable or can keep it hardcoded. 
As far as I know we only plan to submit ETH/USD prices for now. 
@alextes What do you think ?

**Review by ckoopmann — COMMENTED** [review]
Great stuff, thank you so much. Mostly LGTM.
Wondering if we could add some integration testing though to make sure that it is actually compatible with the latest version of gofer. That'd be amazing. 

Also not sure why the CI hasn't been triggered yet. Probably because it is a PR from a Fork. I'm a github noob but maybe @konstantinzolotarev or @alextes know how to trigger it so we can get a green light there before merging.

Note:
We will also have to change the `docker-compose` file on the server side to include running the gofer service: https://github.com/ultrasoundmoney/oracle-server/blob/main/docker-compose.yml

(I'm increasingly regretting not putting client and server into a mono repo😅)

File: docker-compose.yml
Comment (ckoopmann) [reply docker-compose.yml:4]: Also we probably want to adjust the `README` to include instructions on this.

File: src/main.rs
Comment (alextes) [reply src/main.rs]: That the mock server was actually called and not maybe a real server instead.

**Comment by alextes** [comment]
>Also not sure why the CI hasn't been triggered yet. Probably because it is a PR from a Fork. I'm a github noob but maybe @konstantinzolotarev or @alextes know how to trigger it so we can get a green light there before merging.

I read it's a security feature. Someone could fork your repo, update your actions, and then start printing your secrets. I think after conflicts are resolved maybe we may get a button that says "trust this PR and run CI".

I just made some major changes to this repo which break this PR, sorry about that. Little is needed to fix the conflicts, most of the work from main.rs should move to attestation_scheduler.rs but that is about it. If you get complicated errors around the PriceProvider trait being async and it's holding up the PR I'd suggest you push what you have anyway with those errors. Happy to fix them for you (:

Integration test would be nice, if it's too much, just skip it and we can do it.

File: src/price_provider/gofer/mod.rs
Comment (alextes) [reply src/price_provider/gofer/mod.rs]: Yea I think we can keep the ETHUSD pair hardcoded for the foreseeable future. Although I'd be open to having `new` use a hardcoded pair for example but having a second constructor which does allow any other pair to be passed if that helps @konstantinzolotarev in any way.

File: docker-compose.yml
Comment (konstantinzolotarev) [reply docker-compose.yml:4]: Yep it was a quick build, I'll adjust README and docs for it. But yes to start HTTP interface you need to run it using `gofer agent`

File: src/main.rs
Comment (konstantinzolotarev) [reply src/main.rs]: Yes exactly it does check that mock server was called

**Comment by konstantinzolotarev** [comment]
Seems that everything is good after rebasing and applying some fixes to my code. 
Now should be ok, also made some fixes regarding your comments. And squashed commits for cleaner integration process

**Review by alextes — APPROVED** [review]

**Comment by alextes** [comment]
Thank you Konstatin!

**Pull request merged**
